File: Java/Desktop/NewLab8/src/by/bntu/fitr/povt/cng/model/collections/comparators/WeightComparatorToLower.java
package by.bntu.fitr.povt.cng.model.collections.comparators;

import by.bntu.fitr.povt.cng.model.entity.abstracts.Toy;

import java.util.Comparator;

public class WeightComparatorToLower implements Comparator<Toy> {
    @Override
    public int compare(Toy o1, Toy o2) {
        return (int) (o2.getWeight() - o2.getWeight());
    }
}

File: Csharp/Lab8/Lab8/Lab8/cng/Model/Collections/Comparators/WeightComparatorToLower.cs
using System.Collections;
using Lab8.cng.Model.Entity.Abstracts;

namespace Lab8.cng.Model.Collections.Comparators
{
    public class WeightComparatorToLower : IComparer
    {
        public int Compare(object x, object y)
        {
            var toy = (Toy) x;
            var toy1 = (Toy) y;
            return (int) (toy1.GetWeight() - toy.GetWeight());
        }
    }
}